plugins/modules/network/routeros/routeros_api.py

```diff
@@ -0,0 +1,489 @@
+      debug:
+        msg: '{{ print_path }}'
+
+    - name: add "ip address add {{ addips[0] }}" "ip address add {{ addips[1] }}"
+      routeros_api:
+        hostname: "{{ hostname }}"
+        password: "{{ password }}"
+        username: "{{ username }}"
+        path: "{{ path }}"
+        add: "{{ item }}"
+      loop: "{{ addips }}"
+      register: addout
+
+    - name: result routeros '.id' for new added items
+      debug:
+        msg: '{{ addout }}'
+
+    - name: query for ".id" in "{{ path }} WHERE address == {{ ip2 }}"
+      routeros_api:
+        hostname: "{{ hostname }}"
+        password: "{{ password }}"
+        username: "{{ username }}"
+        path: "{{ path }}"
+        query: ".id address WHERE address == {{ ip2 }}"
+      register: queryout
+
+    - name: result query result and set fact with '.id' for {{ ip2 }}
+      debug:
+        msg: '{{ queryout }}'
+
+    - set_fact:
+        query_id : "{{ queryout['msg'][0]['.id'] }}"
+
+    - name: update ".id = {{ query_id }}" taken with custom fact "fquery_id"
+      routeros_api:
+        hostname: "{{ hostname }}"
+        password: "{{ password }}"
+        username: "{{ username }}"
+        path: "{{ path }}"
+        update: ".id={{ query_id }} address={{ ip3 }}"
+      register: updateout
+
+    - name: result prunt update status
+      debug:
+        msg: '{{ updateout }}'
+
+    - name: remove ips -  stage 1 - query for '.id' {{ rmips }}
+      routeros_api:
+        hostname: "{{ hostname }}"
+        password: "{{ password }}"
+        username: "{{ username }}"
+        path: "{{ path }}"
+        query: ".id address WHERE address == {{ item }}"
+      register: id_to_remove
+      loop: "{{ rmips }}"
+
+    # set fact for '.id' from 'query for {{ path }}'
+    - set_fact:
+        to_be_remove: "{{ to_be_remove |default([]) + [item['msg'][0]['.id']] }}"
+      loop: "{{ id_to_remove.results }}"
+
+    - name: remove ips stage 1 - dump '.id'
+      debug:
+        msg: '{{ to_be_remove }}'
+
+    # Remove {{ 'rmips' }} with '.id' by 'to_be_remove' from query
+    - name: remove ips -  stage 2 - remove {{ rmips }} by '.id'
+      routeros_api:
+        hostname: "{{ hostname }}"
+        password: "{{ password }}"
+        username: "{{ username }}"
+        path: "{{ path }}"
+        remove: "{{ item }}"
+      register: remove
+      loop: "{{ to_be_remove }}"
+
+    - name: remove ips stage 2 dump result
+      debug:
+        msg: '{{ remove }}'
+
+    - name: arbitrary command example "/system identity print"
+      routeros_api:
+        hostname: "{{ hostname }}"
+        password: "{{ password }}"
+        username: "{{ username }}"
+        path: "system identity"
+        cmd: "print"
+      register: cmdout
+
+    - name: dump "/system identity print" output
+      debug:
+        msg: "{{ cmdout }}"
+'''
+
+RETURN = '''
+---
+message:
+    description: All outputs are in list with dictionary
+     elements returned from RouterOS api
```

Comment:
```suggestion
    description: All outputs are in list with dictionary elements returned from RouterOS api.
```
we need a dot at the end according to the doc